src/ControlSystem/Measurements/NonFactoryCreatable.hpp

```diff
@@ -0,0 +1,19 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+/*!
+ * \brief Simple struct that changes a class from being factory creatable, to
+ * non-factory creatable.
+ *
+ * It does this by defining and setting a static constexpr bool
+ * `factory_creatable = false`.
+ *
+ * \tparam FactoryCreatableClass Any class that is already factory creatable.
+ */
+template <typename FactoryCreatableClass>
+struct NonFactoryCreatableWrapper : public FactoryCreatableClass {
+  static constexpr bool factory_creatable = false;
+  using FactoryCreatableClass::FactoryCreatableClass;
+};
```

Comment:
I think you need `WRAPPED_PUPable_decl_template(NonFactoryCreatableWrapper)` for serialization to work properly.

Reply:
Currently, these events don't have member data or `pup` functions. I tried adding some member data and a `pup` function to test the serialization, and the serialized (and deserialized) class was the same. I wonder if it's because the `NonFactoryCreatableWrapper` doesn't hold any data, so it just inherits the pup stuff from the `FactoryCreatableClass` and that works ok.

Reply:
I think you run into trouble if both `T` and `NonFactoryCreatableWrapper<T>` are registered with charm.  The wrapper will inherit the pup stuff, but that will include the mapping of PUP_ID to class, so one of them will deserialize to the wrong type when serialized through a base-class pointer.

You don't need a `pup` function, since the inherited version of that is correct.

Reply:
Have any thoughts about a good way to test this? Only thing I can think of is to register both `T` and `NonFactoryCreatableWrapper<T>` with charm and try to serialize both through a `unique_ptr<::Event>`.

Reply:
That is how I would do it.